### PR TITLE
Proposals for alternative process_options function

### PR DESCRIPTION
**Pull request by bavovanachte**

### CODE REVIEW

**Comment by bavovanachte** [comment]
@JasperCraeghs Some proposals for handling comma-separated configurations while still keeping the process_options approach. Goal is mainly to aid discussion. Open to alternative suggestions of course

**Comment by JasperCraeghs** [comment]
Proposal #2 seems good to me and easy to implement.

**Comment by bavovanachte** [comment]
Implemented proposal 2. I'll add unit tests for process_options later today or tomorrow

**Comment by bavovanachte** [comment]
Added tests in 49cb773 and fixed the issues found in the same commit.

File: mlx/directives/attribute_sort_directive.py [hunk 1]
```diff
@@ -54,5 +54,8 @@ def run(self):
         node['filter'] = r"\S+"
         node['sort'] = []
 
-        self.process_options(node, {'sort': [], 'filter': r"\S+"})
+        self.process_options(node,
+                             {'sort':   {'default': []},
+                              'filter': {'default': r"\S+"}
+                              })
```
Comment (JasperCraeghs) [inline mlx/directives/attribute_sort_directive.py:60]:
```suggestion:-2+0
                             {
                                 'sort':   {'default': []},
                                 'filter': {'default': r"\S+"},
                              })
```
I prefer this style of indentation

File: mlx/directives/item_2d_matrix_directive.py [hunk 2]
```diff
@@ -97,11 +97,11 @@ def run(self):
...
-                             {'target': '',
-                              'source': '',
-                              'type': [],
-                              'hit': 'x',
-                              'miss': '',
+                             {'target': {'default': ''},
+                              'source': {'default': ''},
+                              'type':   {'default': []},
+                              'hit':    {'default': 'x'},
+                              'miss':   {'default': ''},
```
Comment (JasperCraeghs) [inline mlx/directives/item_2d_matrix_directive.py:104]: Not a fan of alignment like this because it's not as easy to maintain.

**Comment by bavovanachte** [comment]
I've made some alignment updates in e5dc5dc and 5d5ba9c. If I've misinterpreted, please feel free to change the alignment yourself. I don't have strong preferences on the subject.

File: mlx/directives/item_2d_matrix_directive.py (hunk 2 above)
Comment (bavovanachte) [reply mlx/directives/item_2d_matrix_directive.py]: See e5dc5dc and 5d5ba9c

File: mlx/directives/attribute_sort_directive.py (hunk 1 above)
Comment (bavovanachte) [reply mlx/directives/attribute_sort_directive.py]: See e5dc5dc and 5d5ba9c

**Pull request merged**